src/transfer-manager.ts

```diff
@@ -258,13 +260,11 @@ export class TransferManager {
         {},
         options.passthroughOptions
       );
-      if (options.prefix) {
-        passThroughOptionsCopy.destination = path.join(
-          options.prefix || '',
-          passThroughOptionsCopy.destination || '',
-          file.name
-        );
-      }
+      passThroughOptionsCopy.destination = path.join(
+        options.prefix || '',
+        passThroughOptionsCopy.destination || '',
+        file.name
+      );
```

Comment:
I think I will revert this, now that I understand the original design:
1. If `options.prefix` is provided, set `destination`. This causes `files.download` to download files to local machine 
2. If `options.prefix` is not provided, `destination` is not set. This causes `files.download` to return a promise
3.  If `options.prefix` is not provided AND `options.passThroughOptions` is a `function/callback`, then apply the callback onto the results of the file download.